agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1315,28 +1315,30 @@ func (dg *dockerGoClient) APIVersion() (dockerclient.DockerVersion, error) {
 }
 
 // Stats returns a channel of *types.StatsJSON entries for the container.
-func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeout time.Duration) (<-chan *types.StatsJSON, error) {
+func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeout time.Duration) (<-chan *types.StatsJSON, <-chan error) {
 	subCtx, cancelRequest := context.WithCancel(ctx)
 
 	client, err := dg.sdkDockerClient()
 	if err != nil {
 		cancelRequest()
-		return nil, err
+		return nil, nil
```

Comment:
any reason why we are not creating errC before this and returning with error?

Reply:
because we'd have to spin up a goroutine just to send an error down the channel.

Reply:
right, it just seems a bit counter-intuitive below that we are checking are nils and then returning an error.

Reply:
okay, I can change it to spin up a goroutine